### PR TITLE
#311 - StorageHasMetadata matcher

### PR DESCRIPTION
**Pull request by olenagerasimova**
Created `StorageHasMetadata` to avoid code duplications and used it in `RpmITCase` and `RpmTest`  for #311.

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/rpm-adapter/issues/312#issuecomment-653366053))

@olenagerasimova Job `gh:artipie/rpm-adapter#312` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/rpm-adapter/issues/312#issuecomment-653366053))

@olenagerasimova Pull request [#312](https://github.com/artipie/rpm-adapter/issues/312) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

File: src/test/java/com/artipie/rpm/hm/StorageHasMetadata.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+     * @checkstyle ParameterNumberCheck (10 lines)
+     */
+    private static boolean hasMetadata(
+        final Storage storage, final Path temp, final XmlPackage pckg, final int expected
+    ) throws InterruptedException, IOException {
+        final BlockingStorage bsto = new BlockingStorage(storage);
+        final List<Key> repodata = bsto.list(new Key.From("repodata")).stream()
+            .filter(key -> key.string().contains(pckg.filename())).collect(Collectors.toList());
+        if (repodata.size() != 1) {
+            throw new IOException(
```
Comment (olegmoz) [inline src/test/java/com/artipie/rpm/hm/StorageHasMetadata.java:101]: @olenagerasimova should not this case produce `false` result, indicating that metadata in storage is not correct?

File: src/test/java/com/artipie/rpm/hm/StorageHasMetadata.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    /**
+     * Obtains metadata xml from storage and checks that
+     * this metadata has correct amount of packages.
+     * @param storage Storage
+     * @param temp Temp dir to unpack xml
+     * @param pckg Package type
+     * @param expected Amount of expected items in metadata
+     * @return True if metadata is correct
+     * @throws IOException On IO error
+     * @throws InterruptedException On storage op error
```
Comment (olegmoz) [inline src/test/java/com/artipie/rpm/hm/StorageHasMetadata.java:91]: @olenagerasimova that does not look like correct description for `InterruptedException`, it occurs on interruption of waiting thread. I'd suggest to throw just `Exception` here, they are not getting handled in any special way

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected, please, take a look

**Comment by 0crat** [comment]
Job `gh:artipie/rpm-adapter#312` is not assigned, can't get performer

**Pull request merged**